src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -162,11 +165,12 @@ final case class PurposeApiServiceImpl(
       organizationId <- getOrganizationIdFutureUUID(contexts)
       uuid           <- id.toFutureUUID
       purpose        <- purposeManagementService.getPurpose(uuid)
-      eService       <- catalogManagementService.getEServiceById(purpose.eserviceId)
+      isValidRiskAnalysisForm = isRiskAnalysisFormValid(purpose.riskAnalysisForm)
+      eService <- catalogManagementService.getEServiceById(purpose.eserviceId)
       authorizedPurpose =
         if (organizationId == purpose.consumerId || organizationId == eService.producerId) purpose
         else purpose.copy(riskAnalysisForm = None) // Hide risk analysis to other organizations
```

Comment:
may we move the authorization here?
We risk to make mistakes in future changes if we apply authorization in 2 different places
```suggestion
        if (organizationId == purpose.consumerId || organizationId == eService.producerId) purpose.dependencyToApi(isRiskAnalysisFormValid(purpose.riskAnalysisForm))
        else purpose.copy(riskAnalysisForm = None).dependencyToApi(isRiskAnalysisValid = false) // Hide risk analysis to other organizations
```